modules/friendlytag.js

```diff
@@ -240,6 +240,26 @@ Twinkle.tag.callback = function friendlytagCallback() {
 		}
 	});
 
+	// Hacky handling of form submission when quickfilter hides required fields that weren't filled in:
+	// Remove the form submit handler
+	/** @param {HTMLFormElement} form */
+	var submitForm = function(form) {
+		if (form.checkValidity()) {
+			Twinkle.tag.callback.evaluate({target: form});
+		} else {
+			$(form.quickfilter).val('');
+			// XXX: $(form.quickfilter).keyup() doesn't seem to work, so manually doing what the
+			// keyup handler should do
+			$allCheckboxDivs.show();
+			$allHeaders.show();
+			$(form).find('input[type=submit]').click(); // triggers browser validation which should fail
+		}
+	};
+	result.removeEventListener('submit', Twinkle.tag.callback.evaluate);
+	$('.tw-tag-submit').off('click').on('click', function() {
+		submitForm(result);
```

Comment:
Despite the `off` and `on`, this still is being triggered twice. 🤔

But this doesn't seem to be a problem as  the tagging and all supplementary actions (like adding the merge tag on the other page and posting to WP:PNT) are taking place once only.

Reply:
@siddharthvp I get a very laggy interface and a `too much recursion` error here/at the `.click` line.  I'm seeing it run twice, sure, but each time it looks like these are running 13 times.  FF fwiw.